spec/civiltime.html

```diff
@@ -1,230 +1,558 @@
+        1. Return ? CreateTime(_hour_, _minute_, _second_, _millisecond_, _microsecond_, _nanosecond_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause>
+      <h1>Temporal.Time.prototype.with ( _timelike_ [ , _disambiguation_ ] )</h1>
+      <p>
+        The `with` method takes two arguments, _timelike_ and _disambiguation_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _time_ be the *this* value.
+        1. If Type(_time_) is not Object or _time_ does not have an [[InitializedTemporalTime]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Let _timelike_ be ? ToPartialTime(_timelike_).
+        1. Let _disambiguation_ be ? ToString(_disambiguation_).
+        1. If _timelike_.[[Hour]] is not *undefined*, then
+          1. Let _hour_ be _timelike_.[[Hour]].
+        1. Else
+          1. Let _hour_ be _time_.[[Hour]].
+        1. If _timelike_.[[Minute]] is not *undefined*, then
+          1. Let _minute_ be _timelike_.[[Minute]].
+        1. Else
+          1. Let _minute_ be _time_.[[Minute]].
+        1. If _timelike_.[[Second]] is not *undefined*, then
+          1. Let _second_ be _timelike_.[[Second]].
+        1. Else
+          1. Let _second_ be _time_.[[Second]].
+        1. If _timelike_.[[Millisecond]] is not *undefined*, then
+          1. Let _millisecond_ be _timelike_.[[Millisecond]].
+        1. Else
+          1. Let _millisecond_ be _time_.[[Millisecond]].
+        1. If _timelike_.[[Microsecond]] is not *undefined*, then
+          1. Let _microsecond_ be _timelike_.[[Microsecond]].
+        1. Else
+          1. Let _microsecond_ be _time_.[[Microsecond]].
+        1. If _timelike_.[[Nanosecond]] is not *undefined*, then
+          1. Let _nanosecond_ be _timelike_.[[Nanosecond]].
+        1. Else
+          1. Let _nanosecond_ be _time_.[[Nanosecond]].
+        1. If ! ValidateTime(_hour_, _minute_, _second_, _millisecond_, _microsecond_, _nanosecond_) is *false*, then
+          1. Throw a *TypeError* exception.
+        1. If _disambiguation_ is not *undefined*, then
+          1. Set _disambiguation_ to ? ToString(_disambiguation_).
+        1. Else
+          1. Set _disambiguation_ to `"constrain"`.
+        1. Let _result_ be ? RegulateTime(_hour_, _minute_, _second_, _millisecond_, _microsecond_, _nanosecond_, _disambiguation_).
+        1. Return ? CreateTime(_result_.[[Hour]], _result_.[[Minute]], _result_.[[Second]], _result_.[[Millisecond]], _result_.[[Microsecond]], _result_.[[Nanosecond]]).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause>
+      <h1>Temporal.Time.prototype.difference ( _otherTime_ )</h1>
+      <p>
+        The `difference` method takes one argument _otherTime_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _time_ be the *this* value.
+        1. If Type(_time_) is not Object or _time_ does not have an [[InitializedTemporalTime]] internal slot, then
```

Comment:
this “if” and throw is repeated many times. Perhaps like RequireObjectCoercible, it’d  be helpful to have a RequireTime abstract op?

Reply:
I can do that if people agree it makes sense.

Reply:
https://github.com/tc39/proposal-temporal/issues/207